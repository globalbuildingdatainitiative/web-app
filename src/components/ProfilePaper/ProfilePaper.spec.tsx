import { ProfilePaper } from '@components'
import { render, screen, suite, test } from '@test'
import { MockedProvider } from '@apollo/client/testing'
import { useGetContributionsPerMonthQuery } from '@queries'
import { vi, expect } from 'vitest'

vi.mock('@queries', () => ({
  useGetContributionsPerMonthQuery: vi.fn(),
}))

// Mock data for contributions
const mockContributionsData = {
  contributions: {
    items: [
      { uploadedAt: '2024-08-01T00:00:00Z' },
      { uploadedAt: '2024-08-15T00:00:00Z' },
      { uploadedAt: '2024-09-01T00:00:00Z' },
    ],
    count: 3,
  },
}

;(useGetContributionsPerMonthQuery as jest.Mock).mockReturnValue({
  loading: false,
  error: undefined,
  data: mockContributionsData,
})

suite('ProfilePaper', () => {
  test('Snapshot', () => {
    const { container } = render(
      <MockedProvider>
        <ProfilePaper />
      </MockedProvider>,
    )
    expect(container).toMatchSnapshot()
  })

  test('Render Correctly', async () => {
    render(
      <MockedProvider>
        <ProfilePaper />
      </MockedProvider>,
    )
    expect(await screen.findByTestId('ProfilePaper')).to.exist
  })

  test('Render User Contributions Chart', async () => {
    render(
      <MockedProvider>
        <ProfilePaper />
      </MockedProvider>,
    )
    expect(await screen.findByText('User Contributions Over Time (Monthly)')).to.exist
  })

  test('Render Contributions Data Correctly', async () => {
    render(
      <MockedProvider>
        <ProfilePaper />
      </MockedProvider>,
    )
    expect(await screen.findByText('User Contributions Over Time (Monthly)')).to.exist
  })
})
