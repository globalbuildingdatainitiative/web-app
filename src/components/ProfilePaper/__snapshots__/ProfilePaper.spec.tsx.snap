// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfilePaper > Snapshot 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: Outfit, sans-serif;--mantine-font-family-headings: Outfit, sans-serif;--mantine-breakpoint-xxl: 120em;}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}@media (max-width: 119.99375em) {.mantine-visible-from-xxl {display: none !important;}}@media (min-width: 120em) {.mantine-hidden-from-xxl {display: none !important;}}
  </style>
  <div
    class="m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
    data-testid="ProfilePaper"
    data-with-border="true"
    style="--paper-radius: var(--mantine-radius-lg); --paper-shadow: var(--mantine-shadow-sm); --card-padding: var(--mantine-spacing-lg);"
  >
    <h3
      class="m_8a5d1357 mantine-Title-root"
      data-order="3"
      style="--title-fw: var(--mantine-h3-font-weight); --title-lh: var(--mantine-h3-line-height); --title-fz: var(--mantine-h3-font-size); margin-bottom: 8px;"
    >
      Profile
    </h3>
    <p>
      Here Profile charts will be displayed
    </p>
  </div>
</div>
`;
